integration/spark/spark3/src/main/java/io/openlineage/spark3/agent/utils/PlanUtils3.java

```diff
@@ -67,6 +67,19 @@ public static Optional<DatasetIdentifier> getDatasetIdentifier(
     }
   }
 
+  public static Optional<DatasetIdentifier> getDatasetIdentifierFromRelation(
```

Comment:
What's the benefit of having this as a separate `public` method? From the API perspective, it does the same as `getDatasetIdentifier` method above. Perhaps this could be a private static method called from the `getDatasetIdentifier`

Reply:
Thank you for the feedback - we just updated it like you suggested.

Reply:
@pawel-big-lebowski, one note: updating the getDatasetIdentifier method as you suggested works well, but we do now need to update the following test: 

io.openlineage.spark3.agent.utils.PlanUtils3Test.testGetDatasetIdentifierFromV2RelationWithMissingIdentifier

It assumes that if the identifier is null, we return Optional.empty(), however with the relation handler, we now have the possibility of finding the dataset despite the identifier being null.

Would you suggest we modify the test to reflect the change?